tmt/base.py

```diff
@@ -1223,7 +1248,21 @@ def _save_tree(self, tree):
     def environment(self):
         """ Return environment combined from wake up and command line """
         combined = self._environment.copy()
-        combined.update(tmt.utils.environment_to_dict(self.opt('environment')))
+        environment_vars = tmt.utils.environment_to_dict(
+            self.opt('environment'))
+        environment_file_vars = tmt.utils.environment_file_to_dict(
+            self.opt('environment-file') or []
+            )
+
+        # combine all env sources into one
+        combined.update(**environment_file_vars)
+        combined.update(**environment_vars)
```

Comment:
Is there any advantage from `combined.update(environment_file_vars)`, e.g. not using the double-asterisk?

Reply:
In this case not. It may have sense when doing this:
```python
In [1]: r = {"a": None, "b": None}

In [2]: r.update(**{"a":1}, **{"b":2})

In [3]: r
Out[32]: {'a': 1, 'b': 2}
```

Reply:
resolved